worlds/terraria/__init__.py

```diff
@@ -0,0 +1,329 @@
+            if condition == "npc":
+                if type(arg) is not int:
+                    raise Exception("@npc requires an integer argument")
+
+                npc_count = 0
+                for npc in npcs:
+                    if state.has(npc, self.player):
+                        npc_count += 1
+                        if npc_count >= arg:
+                            return sign
+
+                return not sign
+            elif condition == "calamity":
+                return sign == self.calamity
+            elif condition == "pickaxe":
+                if type(arg) is not int:
+                    raise Exception("@pickaxe requires an integer argument")
+
+                for pickaxe, power in pickaxes.items():
+                    if power >= arg and state.has(pickaxe, self.player):
+                        return sign
+
+                return not sign
+            elif condition == "hammer":
+                if type(arg) is not int:
+                    raise Exception("@hammer requires an integer argument")
+
+                for hammer, power in hammers.items():
+                    if power >= arg and state.has(hammer, self.player):
+                        return sign
+
+                return not sign
+            elif condition == "mech_boss":
+                if type(arg) is not int:
+                    raise Exception("@mech_boss requires an integer argument")
+
+                boss_count = 0
+                for boss in mech_bosses:
+                    if state.has(boss, self.player):
+                        boss_count += 1
+                        if boss_count >= arg:
+                            return sign
+
+                return not sign
+            elif condition == "minions":
+                if type(arg) is not int:
+                    raise Exception("@minions requires an integer argument")
+
+                minion_count = 1
+                for armor, minions in armor_minions.items():
+                    if state.has(armor, self.player) and minions + 1 > minion_count:
+                        minion_count = minions + 1
+                        if minion_count >= arg:
+                            return sign
+
+                for accessory, minions in accessory_minions.items():
+                    if state.has(accessory, self.player):
+                        minion_count += minions
+                        if minion_count >= arg:
+                            return sign
+
+                return not sign
+            else:
+                raise Exception(f"Unknown function {condition}")
+        elif ty == COND_GROUP:
+            operator, conditions = condition
+            return sign == self.check_conditions(state, operator, conditions)
+
+    def check_conditions(
+        self,
+        state,
+        operator,  #: bool | None,
+        conditions,  #: list[tuple[bool, int, str | tuple[bool | None, list], str | int | None]],
+    ) -> bool:
+        if operator is None:
```

Comment:
This feels like it could be done with assert, to be skipped when shipped, as this should only raise during dev, correct?

Reply:
No, `operator` is allowed to be `None` when there's only one condition in the group. Like if some conditions are just `Wall of Flesh`, no `&` or `|` is provided, so `operator` is `None`. As opposed to `Wall of Flesh & Plantera`, where `operator` would be `false` to represent `&`